Mathlib/Order/RelIso/Basic.lean

```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
   ⟨Embedding.quotientOut α, by
     refine' @fun x y => Quotient.inductionOn₂ x y fun a b => _
     apply iff_iff_eq.2 (H _ _ _ _ _ _) <;> apply Quotient.mk_out⟩
 #align quotient.out_rel_embedding Quotient.outRelEmbedding
 #align quotient.out_rel_embedding_apply Quotient.outRelEmbedding_apply
 
+/-- `quotient.out'` as a relation embedding between the lift of a relation and the relation. -/
+@[simps]
+noncomputable def Quotient.out'RelEmbedding {_ : Setoid α} {r : α → α → Prop}
+    (H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂) :
+    (fun a b => Quotient.liftOn₂' a b r H) ↪r r :=
+  { Quotient.outRelEmbedding _ with toFun := Quotient.out' }
+#align rel_embedding.quotient.out'_rel_embedding Quotient.out'RelEmbedding
+
+@[simp]
+theorem acc_lift₂_iff [Setoid α] {r : α → α → Prop}
+    {H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂} {a} :
+    Acc (Quotient.lift₂ r H) ⟦a⟧ ↔ Acc r a :=
+  by
+  constructor
+  · exact RelHomClass.acc (Quotient.mkRelHom H) a
+  · intro ac
+    induction' ac with _ _ IH
+    dsimp at IH
+    refine' ⟨_, fun q h => _⟩
+    obtain ⟨a', rfl⟩ := q.exists_rep
+    exact IH a' h
+#align acc_lift₂_iff acc_lift₂_iff
+
+@[simp]
+theorem acc_liftOn₂'_iff {s : Setoid α} {r : α → α → Prop} {H} {a} :
+    Acc (fun x y => Quotient.liftOn₂' x y r H) (Quotient.mk'' a : Quotient s) ↔ Acc r a :=
+  acc_lift₂_iff
+#align acc_lift_on₂'_iff acc_liftOn₂'_iff
+
 /-- A relation is well founded iff its lift to a quotient is. -/
 @[simp]
-theorem _root_.wellFounded_lift₂_iff [s : Setoid α] {r : α → α → Prop}
+theorem wellFounded_lift₂_iff [s : Setoid α] {r : α → α → Prop}
     {H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂} :
     WellFounded (Quotient.lift₂ r H) ↔ WellFounded r :=
-  ⟨fun hr => by
-    suffices ∀ {x : Quotient s} {a : α}, ⟦a⟧ = x → Acc r a by exact ⟨fun a => this rfl⟩
-    · refine' @fun x => @WellFounded.induction _ _ hr
-         (fun y => ∀ (a : α), Quotient.mk s a = y → Acc r a) x _
-      rintro x IH a rfl
-      exact ⟨_, @fun b hb => IH ⟦b⟧ hb _ rfl⟩
-      ,
-    (Quotient.outRelEmbedding H).wellFounded⟩
+  by
+  constructor
+  · exact RelHomClass.wellFounded (Quotient.mkRelHom H)
+  · refine' fun wf => ⟨fun q => _⟩
+    obtain ⟨a, rfl⟩ := q.exists_rep
+    exact acc_lift₂_iff.2 (wf.apply a)
 #align well_founded_lift₂_iff wellFounded_lift₂_iff
 
 alias wellFounded_lift₂_iff ↔ _root_.WellFounded.of_quotient_lift₂ _root_.WellFounded.quotient_lift₂
 
+@[simp]
+theorem wellFounded_liftOn₂'_iff {s : Setoid α} {r : α → α → Prop} {H} :
+    (WellFounded fun x y : Quotient s => Quotient.liftOn₂' x y r H) ↔ WellFounded r :=
+  wellFounded_lift₂_iff
+#align well_founded_lift_on₂'_iff wellFounded_liftOn₂'_iff
+
+alias wellFounded_liftOn₂'_iff ↔
+  wellFounded.of_quotient_liftOn₂' wellFounded.quotient_liftOn₂'
+#align well_founded.of_quotient_lift_on₂' wellFounded.of_quotient_liftOn₂'
+#align well_founded.quotient_lift_on₂' wellFounded.quotient_liftOn₂'
```

Comment:
The mathport output here was correct, but it seems you edited it to be incorrect.

Reply:
Sorry, I don't quite get the capitalization rules. Is `WellFounded` always uppercase like that?

Reply:
Well you're trying to put these lemmas in the namespace that matches the type, right? And the `wellFounded` namespace doesn't exist.